### PR TITLE
Add docs for `V1 Data Engine` and `Guaranteed Instance Manager CPU for V2 Data Engine`

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#7361

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM.

File: content/docs/1.6.0/references/settings.md [hunk 1]
```diff
@@ -810,7 +830,7 @@ See [Trim Filesystem](../../volumes-and-nodes/trim-filesystem) for details.
 
 > Default: `12`
 
-This integer value indicates how many percentage of the total allocatable CPU on each node will be reserved for each instance manager Pod. For example, 10 means 10% of the total CPU on a node will be allocated to each instance manager pod on this node. This will help maintain engine and replica stability during high node workload.
+Percentage of the total allocatable CPU resources on each node to be reserved for each instance manager pod when the V1 Data Engine is enabled. For example, 10 means 10% of the total CPU on a node will be allocated to each instance manager pod on this node. This will help maintain engine and replica stability during high node workload.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/references/settings.md:833]:
```suggestion
Percentage of the total allocatable CPU resources on each node to be reserved for each instance manager pod when the V1 Data Engine is enabled. For example, Longhorn reserves 10% of the total allocatable CPU resources if you specify a value of 10. This setting is essential for maintaining engine and replica stability, especially during periods of high node workload.
```

File: content/docs/1.6.0/references/settings.md [hunk 2]
```diff
@@ -448,6 +457,17 @@ This allows users to activate the v2 data engine based on SPDK. Currently, it is
...
+#### Guaranteed Instance Manager CPU for V2 Data Engine
+
+> Default: `1250`
+
+Number of millicpus on each node to be reserved for each instance manager pod when the V2 Data Engine is enabled. By default, the Storage Performance Development Kit (SPDK) target daemon within each instance manager pod uses 1 CPU core. Configuring a minimum CPU usage value is essential for maintaining engine and replica stability, especially during periods of high node workload.
+
+> **Warning:**
+>  - Value 0 means unsetting CPU requests for instance manager pods for v2 data engine.
+>  - This integer value is range from 1000 to 8000.
+>  - After this setting is changed, all instance manager pods using this global setting on all the nodes will be automatically restarted. In other words, DO NOT CHANGE THIS SETTING WITH ATTACHED VOLUMES.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/references/settings.md:469]:
```suggestion
>  - Specifying a value of 0 disables CPU requests for instance manager pods. You must specify an integer between 1000 and 8000. 
>  - This is a global setting. Modifying the value triggers an automatic restart of the instance manager pods. Do not modify the value while volumes are still attached.
```

File: content/docs/1.6.0/best-practices.md [hunk 3]
```diff
@@ -117,14 +120,20 @@ Longhorn system will create snapshots automatically when rebuilding a replica. R
...
 
 You can also set a specific milli CPU value for instance manager pods on a particular node by updating the node's `Instance Manager CPU Request` field.
 
 > **Note:** This field will overwrite the above setting for the specified node.
 
 Refer to [Guaranteed Instance Manager CPU](../references/settings/#guaranteed-instance-manager-cpu) for more details.
 
+### V2 Data Engine
+
+The `Guaranteed Instance Manager CPU for V2 Data Engine` setting allows user to reserve number of millicpus on each node for each instance manager pod when the V2 Data Engine is enabled. By default, the Storage Performance Development Kit (SPDK) target daemon within each instance manager pod uses 1 CPU core. Configuring a minimum CPU usage value is essential for maintaining engine and replica stability, especially during periods of high node workload. The default value is 1250 millicpus.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/best-practices.md:135]:
```suggestion
The `Guaranteed Instance Manager CPU for V2 Data Engine` setting allows you to reserve a specific number of millicpus on each node for each instance manager pod when the V2 Data Engine is enabled. By default, the Storage Performance Development Kit (SPDK) target daemon within each instance manager pod uses 1 CPU core. Configuring a minimum CPU usage value is essential for maintaining engine and replica stability, especially during periods of high node workload. The default value is 1250.
```

File: content/docs/1.6.0/best-practices.md [hunk 4]
```diff
@@ -117,14 +120,20 @@ Longhorn system will create snapshots automatically when rebuilding a replica. R
 
 We recommend setting the CPU request for Longhorn instance manager pods.
 
-The `Guaranteed Instance Manager CPU` setting allows you to reserve a percentage of a node's total allocatable CPU for all instance manager pods.
+### V1 Data Engine
+
+The `Guaranteed Instance Manager CPU` setting allows user to reserver percentage of the total allocatable CPU resources on each node for each instance manager pod when the V1 Data Engine is enabled. The default value is 12 percent.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/best-practices.md:125]:
```suggestion
The `Guaranteed Instance Manager CPU` setting allows you to reserve a percentage of the total allocatable CPU resources on each node for each instance manager pod when the V1 Data Engine is enabled. The default value is 12.
```

File: content/docs/1.6.0/references/helm-values.md [hunk 5]
```diff
@@ -225,12 +225,13 @@ For more details like types or options, you can refer to **Settings Reference**
 | defaultSettings.engineReplicaTimeout | In seconds. The setting specifies the timeout between the engine and replica(s), and the value should be between 8 to 30 seconds. The default value is 8 seconds. |
 | defaultSettings.failedBackupTTL | In minutes. This setting determines how long Longhorn will keep the backup resource that was failed. Set to 0 to disable the auto-deletion. |
 | defaultSettings.fastReplicaRebuildEnabled | This feature supports the fast replica rebuilding. It relies on the checksum of snapshot disk files, so setting the snapshot-data-integrity to **enable** or **fast-check** is a prerequisite. |
-| defaultSettings.guaranteedInstanceManagerCPU | This integer value indicates how many percentage of the total allocatable CPU on each node will be reserved for each instance manager Pod. You can leave it with the default value, which is 12%. |
+| defaultSettings.guaranteedInstanceManagerCPU | Percentage of the total allocatable CPU resources on each node to be reserved for each instance manager pod when the V1 Data Engine is enabled. The default value is 12 percent. |
```
Comment (jillian-maroket) [inline content/docs/1.6.0/references/helm-values.md:228]:
```suggestion
| defaultSettings.guaranteedInstanceManagerCPU | Percentage of the total allocatable CPU resources on each node to be reserved for each instance manager pod when the V1 Data Engine is enabled. The default value is 12. |
```

File: content/docs/1.6.0/references/helm-values.md [hunk 6]
```diff
@@ -225,12 +225,13 @@ For more details like types or options, you can refer to **Settings Reference**
...
 | defaultSettings.kubernetesClusterAutoscalerEnabled | Enabling this setting will notify Longhorn that the cluster is using Kubernetes Cluster Autoscaler. |
 | defaultSettings.logLevel | The log level Panic, Fatal, Error, Warn, Info, Debug, Trace used in longhorn manager. Default to Info. |
 | defaultSettings.nodeDownPodDeletionPolicy | Defines the Longhorn action when a Volume is stuck with a StatefulSet/Deployment Pod on a node that is down. |
 | defaultSettings.nodeDrainPolicy | Define the policy to use when a node with the last healthy replica of a volume is drained. |
 | defaultSettings.offlineReplicaRebuilding | This setting allows users to enable the offline replica rebuilding for volumes using v2 data engine. |
+| defaultSettings.v2DataEngineGuaranteedInstanceManagerCPU | Number of millicpus on each node to be reserved for each instance manager pod when the V2 Data Engine is enabled. The default value is 1250 millicpus. |
```
Comment (jillian-maroket) [inline content/docs/1.6.0/references/helm-values.md:234]:
```suggestion
| defaultSettings.v2DataEngineGuaranteedInstanceManagerCPU | Number of millicpus on each node to be reserved for each instance manager pod when the V2 Data Engine is enabled. The default value is 1250. |
```

File: content/docs/1.6.0/best-practices.md (hunk 4 above)
Comment (jillian-maroket) [reply content/docs/1.6.0/best-practices.md:125]: When we mention default values, we are referring to the actual user input. In this case, the user is only required to type a number (without the % symbol).

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

**Review by jillian-maroket — APPROVED** [review]

**Pull request merged**